File: smoke/java/src/main/java/Smoke.java
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.HashMap;
import java.util.Map;
import java.util.Optional;
import resources.responses.requests.CreateResponseRequest;
import types.InputMessage;
import types.InputMessageRole;
import types.TextResponseConfig;
import types.TextResponseConfigFormat;

/** Live smoke: one POST /v1/responses via Fern-generated Java SDK sources. */
public class Smoke {

  private static Map<String, String> loadDotenv(Path path) throws Exception {
    Map<String, String> out = new HashMap<>();
    if (!Files.isRegularFile(path)) {
      return out;
    }
    for (String line : Files.readAllLines(path)) {
      String t = line.trim();
      if (t.isEmpty() || t.startsWith("#")) {
        continue;
      }
      int eq = t.indexOf('=');
      if (eq <= 0) {
        continue;
      }
      String k = t.substring(0, eq).trim();
      String v = t.substring(eq + 1).trim();
      if (v.length() >= 2
          && ((v.startsWith("\"") && v.endsWith("\"")) || (v.startsWith("'") && v.endsWith("'")))) {
        v = v.substring(1, v.length() - 1);
      }
      if (!k.isEmpty()) {
        out.put(k, v);
      }
    }
    return out;
  }

  private static String pick(Map<String, String> dotenv, String key) {
    String v = System.getenv(key);
    if (v != null && !v.trim().isEmpty()) {
      return v.trim();
    }
    return Optional.ofNullable(dotenv.get(key)).orElse("").trim();
  }

  public static void main(String[] args) throws Exception {
    Path smokeJava = Path.of(System.getProperty("user.dir"));
    Path fernSdk = smokeJava.getParent().getParent();
    Path workspace = fernSdk.getParent();

    Map<String, String> dotenv = new HashMap<>();
    dotenv.putAll(loadDotenv(smokeJava.resolve(".env")));
    for (Map.Entry<String, String> e : loadDotenv(workspace.resolve("Benchmark").resolve(".env")).entrySet()) {
      dotenv.putIfAbsent(e.getKey(), e.getValue());
    }

    String apiKey = pick(dotenv, "ZEROGPU_API_KEY");
    String projectId = pick(dotenv, "ZEROGPU_PROJECT_ID");
    String model = pick(dotenv, "ZEROGPU_MODEL");
    if (apiKey.isEmpty() || projectId.isEmpty()) {
      System.err.println("Set ZEROGPU_API_KEY and ZEROGPU_PROJECT_ID");
      System.exit(1);
    }
    if (model.isEmpty()) {
      System.err.println("Set ZEROGPU_MODEL");
      System.exit(1);
    }

    String prompt = pick(dotenv, "ZEROGPU_INPUT_TEXT");
    if (prompt.isEmpty()) {
      prompt = "In one short sentence, what is a habit tracker?";
    }

    ZerogpuApiClient client =
        ZerogpuApiClient.builder().apiKey(apiKey).projectId(projectId).build();

    CreateResponseRequest req =
        CreateResponseRequest.builder()
            .model(model)
            .addInput(
                InputMessage.builder()
                    .role(InputMessageRole.USER)
                    .content(prompt)
                    .build())
            .text(
                TextResponseConfig.builder()
                    .format(TextResponseConfigFormat.builder().type("text").build())
                    .build())
            .build();

    var res = client.responses().createResponse(req);
    System.out.println("id: " + res.getId());
    System.out.println("model: " + res.getModel());
    System.out.println("usage: " + res.getUsage());
  }
}

File: smoke/php/smoke.php
<?php

declare(strict_types=1);

/**
 * Live smoke: one POST /v1/responses via Fern-generated PHP SDK.
 */
require __DIR__ . '/vendor/autoload.php';

use Zerogpu\Types\InputMessage;
use Zerogpu\Types\InputMessageRole;
use Zerogpu\Responses\Requests\CreateResponseRequest;
use Zerogpu\Types\TextResponseConfig;
use Zerogpu\Types\TextResponseConfigFormat;
use Zerogpu\ZerogpuClient;

function load_dotenv(string $path): void
{
    if (!is_readable($path)) {
        return;
    }
    foreach (file($path, FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES) ?: [] as $line) {
        $line = trim($line);
        if ($line === '' || str_starts_with($line, '#')) {
            continue;
        }
        if (!str_contains($line, '=')) {
            continue;
        }
        [$k, $v] = explode('=', $line, 2);
        $k = trim($k);
        $v = trim($v, " \t\"'");
        if (($k !== '') && getenv($k) === false) {
            putenv("$k=$v");
            $_ENV[$k] = $v;
        }
    }
}

$here = __DIR__;
$fern = dirname($here, 2);
load_dotenv("$here/.env");
load_dotenv(dirname($fern) . '/Benchmark/.env');

$apiKey = trim((string) getenv('ZEROGPU_API_KEY'));
$projectId = trim((string) getenv('ZEROGPU_PROJECT_ID'));
$model = trim((string) getenv('ZEROGPU_MODEL'));
if ($apiKey === '' || $projectId === '') {
    fwrite(STDERR, "Set ZEROGPU_API_KEY and ZEROGPU_PROJECT_ID\n");
    exit(1);
}
if ($model === '') {
    fwrite(STDERR, "Set ZEROGPU_MODEL\n");
    exit(1);
}

$prompt = trim((string) getenv('ZEROGPU_INPUT_TEXT'));
if ($prompt === '') {
    $prompt = 'In one short sentence, what is a habit tracker?';
}

$client = new ZerogpuClient($apiKey, $projectId);

$req = new CreateResponseRequest([
    'model' => $model,
    'input' => [
        new InputMessage([
            'role' => InputMessageRole::User,
            'content' => $prompt,
        ]),
    ],
    'text' => new TextResponseConfig([
        'format' => new TextResponseConfigFormat([
            'type' => 'text',
        ]),
    ]),
]);

$res = $client->responses->createResponse($req);
if ($res === null) {
    fwrite(STDERR, "Unexpected null response\n");
    exit(1);
}

echo 'id: ' . $res->id . PHP_EOL;
echo 'model: ' . $res->model . PHP_EOL;
echo 'usage: ' . json_encode($res->usage) . PHP_EOL;
